exec-service-wasmer/src/wasmer_metering.rs

```diff
@@ -0,0 +1,197 @@
+use crate::get_opcode_cost;
+use elrond_exec_service::OpcodeCost;
+use loupe::{MemoryUsage, MemoryUsageTracker};
+use std::mem;
+use std::sync::Mutex;
+use wasmer::wasmparser::{Operator, Type as WpType, TypeOrFuncType};
+use wasmer::{
+    ExportIndex, FunctionMiddleware, GlobalInit, GlobalType, Instance, LocalFunctionIndex,
+    MiddlewareError, MiddlewareReaderState, ModuleMiddleware, Mutability, Type,
+};
+use wasmer_types::{GlobalIndex, ModuleInfo};
+
+#[derive(Clone, Debug, MemoryUsage)]
+struct MeteringGlobalIndexes(GlobalIndex, GlobalIndex);
+
+impl MeteringGlobalIndexes {
+    fn points_limit(&self) -> GlobalIndex {
+        self.0
+    }
+
+    fn points_used(&self) -> GlobalIndex {
+        self.1
+    }
+}
+
+#[derive(Debug)]
+pub struct Metering {
+    points_limit: u64,
+    opcode_cost: OpcodeCost,
+    global_indexes: Mutex<Option<MeteringGlobalIndexes>>,
```

Comment:
This Mutex might give us performance issues, we'll need to keep an eye on it. We can leave it like this for now.

Reply:
Yes, my first thought was to also remove it. As discussed we can try to use Arc here instead.